source/common/modules/markdown-editor/hooks/autocomplete.ts

```diff
@@ -324,6 +327,21 @@ function shouldBeginAutocomplete (cm: CodeMirror.Editor, changeObj: any): keyof
     return 'snippets'
   }
 
+  // Can we begin autocompleting a latex command?
+  if (changeObj.text[0] === '\\') {
+    // Check if we are in an equation
+    const lines: LineInfo[] = []
+    for (let i = 0; i < cm.lineCount(); i++) {
```

Comment:
Are there any performance penalties when iterating over long documents in this regard?

Reply:
> Maybe in the future it would make sense to have one markdown parser, that takes care of all parsing (probably using a state-dependent algorithm instead of regex) and keeps an up-to-date representation of the markdown code as a parsed object. Then all the editor commands and syntax highlighting etc could query this central information without the need to reparse the source code.

Sounds like a clever idea. Maybe we can bodge the CodeMirror 6 Lezer for that